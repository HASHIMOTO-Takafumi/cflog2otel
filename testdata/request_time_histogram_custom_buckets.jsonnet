local cel = std.native('cel');
local generate_interval_array(start, end, step) =
  std.map(
    function(x) x * step + start,
    std.range(0, (end - start) / step)
  );

{
  otel: {
    endpoint: 'http://localhost:4317/',
    gzip: true,
  },
  resource_attributes: [
    {
      key: 'service.name',
      value: 'Amazon CloudFront',
    },
    {
      key: 'aws.cloudfront.distribution_id',
      value: cel('cloudfront.distributionId'),
    },
  ],
  scope: {
    name: 'test',
  },
  metrics: [
    {
      name: 'http.server.request_time',
      description: 'The request time of HTTP requests',
      type: 'Histogram',
      interval: '5m',
      unit: 'ms',
      value: cel('log.timeTaken * 1000.0'),
      boundaries: generate_interval_array(0, 10, 2),
      is_cumulative: true,
    },
  ],
}
