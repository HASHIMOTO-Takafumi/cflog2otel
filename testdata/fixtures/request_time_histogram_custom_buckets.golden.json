{
  "Resource": [
    {
      "Key": "aws.cloudfront.distribution_id",
      "Value": {
        "Type": "STRING",
        "Value": "EMLARXS9EXAMPLE"
      }
    },
    {
      "Key": "service.name",
      "Value": {
        "Type": "STRING",
        "Value": "Amazon CloudFront"
      }
    }
  ],
  "ScopeMetrics": [
    {
      "Scope": {
        "Name": "test",
        "Version": "",
        "SchemaURL": ""
      },
      "Metrics": [
        {
          "Name": "http.server.request_time",
          "Description": "The request time of HTTP requests",
          "Unit": "ms",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [],
                "StartTime": "2019-12-04T21:00:00Z",
                "Time": "2019-12-04T21:05:00Z",
                "Count": 3,
                "Bounds": [
                  0,
                  2,
                  4,
                  6,
                  8,
                  10
                ],
                "BucketCounts": [
                  0,
                  3,
                  0,
                  0,
                  0,
                  0,
                  0
                ],
                "Min": 0,
                "Max": 1,
                "Sum": 2
              },
              {
                "Attributes": [],
                "StartTime": "2019-12-13T22:35:00Z",
                "Time": "2019-12-13T22:40:00Z",
                "Count": 3,
                "Bounds": [
                  0,
                  2,
                  4,
                  6,
                  8,
                  10
                ],
                "BucketCounts": [
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  3
                ],
                "Min": 102,
                "Max": 107,
                "Sum": 312
              }
            ],
            "Temporality": "CumulativeTemporality"
          }
        }
      ]
    }
  ]
}